RELEASE_STEPS.md

```diff
@@ -0,0 +1,63 @@
+## UMF Release Steps
+
+This document contains all the steps required to make a new release of UMF.
+
+As a helper, we use in this guide these 2 variables:
+```bash
+  set $VERSION = new full version (e.g., 0.1.0-rc1) # -rc1 included just as an example
+  set $VER = new major+minor only version (e.g., 0.1)
+```
+
+## Make a release locally
+
+Before doing the final release it's recommended to prepare a pre-release version - a "release candidate"
+(or "rc" in short). This requires adding, e.g., `-rc1` to the VERSION string. When all tests and checks
+end properly, you can follow up with the final release. If any fix is required, it should be included in
+another rc package (e.g., `-rc2`).
+
+To do a release:
+- If this is a patch release do if from appropriate "stable" branch, switch to it:
+  - git checkout -b v$VER.x (e.g., `v0.1.x`)
+- Add an entry to ChangeLog, remember to change the day of the week in the release date
+  - For major releases mention API and ABI compatibility with the previous release
+- Update project's version in a few places:
+  - set the new $VERSION in `project` function in the top-level `CMakeLists.txt`
+  - `release` variable in `scripts/docs_config/conf.py` (for docs)
+  - `UMF_VERSION` variable in `.github/workflows/basic.yml` (for installation test)
+- For major releases update ABI version in `.map` and `.def` files
+  - these files are defined for all public libraries (`libumf` and `proxy_lib`, at the moment)
+- Commit these changes and tag the release:
+  - git commit -a -S -m "$VERSION release"
```

Comment:
You could add bash format here and in other places with git commands. I think it will make it clearer to read:
```bash
$ git commit -a -S -m "$VERSION release"
```